src/main/java/net/snowflake/hivemetastoreconnector/SnowflakeConf.java

```diff
@@ -138,6 +138,34 @@ public boolean isSnowflakeJDBCProperty()
 
   private static URL snowflakeConfigUrl;
 
+  /**
+   * Convenience method. Retrieves a secret from a credential provider,
+   * configuration, or default value in the respective order of preference.
+   * @param name the name of the configuration
+   * @param defaultValue the default value of the secret, if any
+   * @return the secret
+   */
+  public String getSecret(String name, String defaultValue)
```

Comment:
Nit: we only ever pass in null as the default value, so we don't really need two parameters. Below you could change get(name, defaultValue) -> get(name) which returns null if the property is not found.